lib/cicognara/catalogo_item.rb

```diff
@@ -22,52 +22,57 @@ def solr_doc
 
     private
 
-    def resolve_avail(arr)
-      arr = (arr || []).flatten.uniq
-      arr.length == 1 ? arr : arr.reject { |v| v == 'None' }
-    end
-
-    def corresp
-      (entry.corresp + books.map(&:digital_cico_number)).uniq
-    end
+      def resolve_avail(arr)
+        arr = (arr || []).flatten.uniq
+        arr.length == 1 ? arr : arr.reject { |v| v == 'None' }
+      end
 
-    def doc_tei_fields
-      { id: n, alt_id: n, cico_s: n, cico_sort: n, tei_description_unstem_search: text, tei_section_display: section_display,
-        tei_section_head_italian: section_head, tei_section_number_display: section_number,
-        tei_author_txt: item_authors, tei_pub_txt: item_pubs, tei_date_display: item_dates,
-        tei_note_italian: item_notes, tei_title_txt: item_titles, tei_section_facet: section_display }
-    end
+      def corresp
+        dlc_numbers = books.map(&:digital_cico_numbers)
+        (entry.corresp + dlc_numbers.flatten).uniq
+      end
 
-    def solr_title_display(label)
-      "Catalogo Item #{label.chomp('.')}"
-    end
+      def doc_tei_fields
+        { id: n, alt_id: n, cico_s: n, cico_sort: n, tei_description_unstem_search: text, tei_section_display: section_display,
+          tei_section_head_italian: section_head, tei_section_number_display: section_number,
+          tei_author_txt: item_authors, tei_pub_txt: item_pubs, tei_date_display: item_dates,
+          tei_note_italian: item_notes, tei_title_txt: item_titles, tei_section_facet: section_display }
+      end
 
-    def marc_fields
-      book_fields = {}
-      books.each do |book|
-        book_doc = book.to_solr
-        book_fields.merge!(book_doc) { |_, oldval, newval| (newval + oldval).uniq } unless book_doc.nil?
+      def solr_title_display(label)
+        "Catalogo Item #{label.chomp('.')}"
       end
-      # fields to ignore
-      remove_display_fields(book_fields)
 
-      # single-valued fields
-      field_first_value(book_fields)
+      def marc_fields
+        book_fields = {}
+        books.each do |book|
+          book_doc = book.to_solr
+          next if book_doc.nil?
+          book_fields.merge!(book_doc) do |_, oldval, newval|
+            values = Array.wrap(newval) + Array.wrap(oldval)
```

Comment:
This also is not properly merging MARC fields extracted from more than one book (in cases where Entries link to more than one Book)